azkaban-common/src/main/java/azkaban/jobtype/JobTypeManager.java

```diff
@@ -386,16 +386,20 @@ public JobParams createJobParams(final String jobId, Props jobProps, final Logge
       }
 
       // inject cluster jars and native libraries into jobs through properties
-      // User's job props should take precedence over cluster props
-      Props finalProps = getClusterSpecificJobProps(targetCluster, jobProps, pluginLoadProps);
-      Props nonOverriddableClusterProps = getClusterSpecificNonOverridableJobProps(finalProps);
-      finalProps.putAll(jobProps);
+      Props clusterSpecificProps = getClusterSpecificJobProps(targetCluster, jobProps, pluginLoadProps);
+      for (final String k : clusterSpecificProps.getKeySet()) {
```

Comment:
please use full name "key" instead of k

Reply:
addressed. and also added an explicit unit test